[4.2] Small fixes in MFA

Kostelano opened the pull request:
Pull Request for Issue # .

### Summary of Changes
1. Fix `showon="mfaredirectonlogin:1"` for **Onboard new users** / **Custom redirection URL parameters**.
2. In the database, I accidentally noticed that the new MFA uses the old language constant, and even from the frontend. Replaced with the correct string.

![Screenshot_1](https://user-images.githubusercontent.com/8440661/177016183-ba9fd8ee-67da-4851-beb7-a44e3c136496.png)

3. Replaced `COM_USERS_HEADING_MFA` with `COM_USERS_USER_MULTIFACTOR_AUTH`, because the first key is also the heading in the users table. For many languages, it makes sense to reduce it to **MFA**, so as not to inflate an already large table. However, the same key was used on the authentication page when logging in - the short **MFA** is inappropriate there, the full name is needed.
4. Language string `PLG_MULTIFACTORAUTH_EMAIL_MAIL_LBL="Joomla: Authentication Code by Email"` was never used.
5. ~~For 2 methods, the `'placeholder'` elements were missing when authenticating on the site (after setting up the methods).~~

Kostelano committed:
Removed unused language strings

Kostelano committed:
Fix "showon" for "mfaredirectonlogin"

Kostelano committed:
Fix header in BackupcodesModel.php

Kostelano committed:
Replaced title, which "can not be shortened"

It is used as the page title for authentication. But also the old header was the header in the users table and in many languages (including Russian, which I'm currently working on) it makes sense to "shorten" it to MFA.

Kostelano committed:
Missing 'placeholder' element for totp and yubikey

Kostelano committed:
Merge branch '4.2-dev' into 4-2-mfa

laoneo commented:
@nikosdion can you have a look here?

nikosdion commented on administrator/language/en-GB/plg_multifactorauth_email.ini:
```diff
@@ -21,7 +21,6 @@ PLG_MULTIFACTORAUTH_EMAIL_LBL_LABEL="Authentication Code"
 PLG_MULTIFACTORAUTH_EMAIL_LBL_PRE_MESSAGE="You have received a six digit Multi-factor Authentication code in your email. Please enter it below."
 PLG_MULTIFACTORAUTH_EMAIL_LBL_SETUP_PLACEHOLDER="Six Digit Authentication Code"
 PLG_MULTIFACTORAUTH_EMAIL_LBL_SHORTINFO="Receive six digit codes by email."
-PLG_MULTIFACTORAUTH_EMAIL_MAIL_LBL="Joomla: Authentication Code by Email"
```
Are you sure this is not used? We have a mail template in System, Mail Templates. This is the long title for the mail template appearing in the Mail Templates list view.

Kostelano replied:
Can you provide a more precise location? I look at the list - I do not see. I also checked through the "phrase search" with third-party software - I don't see it either.

![Screenshot_1](https://user-images.githubusercontent.com/8440661/177053489-c5c919e6-2312-41c8-a4b3-51435a589b06.png)

nikosdion replied:
I may be misremembering. Mark this as resolved.

nikosdion commented on plugins/multifactorauth/totp/src/Extension/Totp.php:
```diff
@@ -138,7 +138,7 @@ public function onUserMultifactorCaptive(Captive $event): void
                     ],
                     // Placeholder text for the HTML input box. Leave empty if you don't need it.
-                    'placeholder'      => '',
+                    'placeholder'      => Text::_('PLG_MULTIFACTORAUTH_TOTP_LBL_SETUP_PLACEHOLDER'),
```
You and Brian asked me NOT to include a placeholder, instead putting explanatory text underneath the field. Therefore I do not accept this change.

Kostelano replied:
No problem. Now we have an element in 2 plugins, and it is absent in others. Maybe I asked (honestly, I don’t remember already), but I made the change so that it was “the same”.

nikosdion commented on plugins/multifactorauth/yubikey/src/Extension/Yubikey.php:
```diff
@@ -136,7 +136,7 @@ public function onUserMultifactorCaptive(Captive $event): void
                     'input_type'         => 'text',
                     // Placeholder text for the HTML input box. Leave empty if you don't need it.
-                    'placeholder'        => '',
+                    'placeholder'        => Text::_('PLG_MULTIFACTORAUTH_YUBIKEY_LBL_SETUP_PLACEHOLDER'),
```
Same as the other placeholder.

Kostelano committed:
Merge branch 'joomla:4.2-dev' into 4-2-mfa

Kostelano committed:
Revert change regarding 'placeholder'

Kostelano committed:
Merge branch '4.2-dev' into 4-2-mfa

Kostelano committed:
Merge branch 'joomla:4.2-dev' into 4-2-mfa

Kostelano committed:
Merge branch '4.2-dev' into 4-2-mfa

Kostelano committed:
Merge branch '4.2-dev' into 4-2-mfa

Kostelano committed:
Merge branch 'joomla:4.2-dev' into 4-2-mfa

heelc29 commented:
@Kostelano Could you please also correct the lang key in the description here:
https://github.com/joomla/joomla-cms/blob/a84ef004541e63ea1952fb9c1ba024265ca2894e/administrator/language/en-GB/com_users.ini#L64
![image](https://user-images.githubusercontent.com/66922325/180315461-6345da6a-3eaf-41d0-93b8-77cede8cb95f.png)

Kostelano committed:
MFA header + COM_MODULES_NONE

https://github.com/joomla/joomla-cms/pull/38208#issuecomment-1191938163 + https://github.com/joomla/joomla-cms/pull/37912#issuecomment-1171384679

Kostelano committed:
Merge branch '4.2-dev' into 4-2-mfa

Kostelano commented:
@heelc29 Thanks, corrected.

Also fixed the display of the **NONE** position for modules - the line is duplicated.

This can happen when you install a large package of extensions, but it is not at all the fact that you will use all the modules, say. In this case, they will be in the **NONE** position. This can confuse users.

![Screenshot_1](https://user-images.githubusercontent.com/8440661/180380513-990325c9-50b2-4c08-9825-dd19aae5b20c.png)

At this point I have everything for this PR. Please test.

heelc29 commented:
I have tested this item :white_check_mark: successfully on 3ecd7cf98a7a3b5af8288bb202a33ce425499675<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38208">issues.joomla.org/tracker/joomla-cms/38208</a>.</sub>

viocassel commented:
I have tested this item :white_check_mark: successfully on 3ecd7cf98a7a3b5af8288bb202a33ce425499675<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38208">issues.joomla.org/tracker/joomla-cms/38208</a>.</sub>

Quy commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38208">issues.joomla.org/tracker/joomla-cms/38208</a>.</sub>

Kostelano committed:
Merge branch 'joomla:4.2-dev' into 4-2-mfa

laoneo committed:
Merge branch '4.2-dev' into 4-2-mfa

roland-d committed:
Merge branch '4.2-dev' into 4-2-mfa

Kostelano committed:
Merge branch '4.2-dev' into 4-2-mfa

fancyFranci committed:
Merge branch '4.2-dev' into 4-2-mfa

fancyFranci committed:
Merge branch '4.2-dev' into 4-2-mfa

fancyFranci commented:
Thanks a lot for your work!
